### DISCUSSION TITLE
Refetch post-Hydration Only? – Next Server-Side SWR Strategy

### DISCUSSION THREAD

**Discussion by sjc5** [TanStack/query#4051] [Q&A] [answered]
Hi there,

Huge RQ fan here, thank you very much for all the work and dedication put into it.  Have a quick question.

I'm working on a Next app that uses RQ server-side to prefetch certain queries, as described in the docs on Next SSR.

It's been working really nicely to set a very generous stale-while-revalidate cache header on the Next handlers, and as long as `staleTime` is set to `0` (i.e., left at default), there is a successful "second" fetch to revalidate data post-hydration (which is desired behavior given the liberal SWR cache headers). 

However, we really don't need such a short staleTime after the initial page load, which made me wonder if there is a clean way to *always* revalidate queries a single time after initial page load / hydration (regardless of staleTime), but *not* every time a component is mounted (assuming data is not stale).

In my testing trying to find a way to make this work, I could only get the queries to refetch post-hydration by either (A) leaving `staleTime` at `0` or (B) setting `refetchOnMount` to `always`, which defeats the purpose of staleTime altogether as far as I can tell.

Thanks in advance for any guidance, and if this isn't possible, I'd love to see it be considered as a future option (if not a default).

Sam

_Aside_: The [useSWR docs](https://swr.vercel.app/docs/with-nextjs#pre-rendering-with-default-data) lead me to believe that this is how their SSR/hydration works by default, but I cannot confirm this in practice as I have only used it in a pure CSR context.

**Comment by sjc5** [comment, marked as answer]
After sleeping on it, it dons on me that perhaps it's really as simple as this?

```
useEffect(() => queryClient.invalidateQueries(), [])
```

**Reply by TkDodo** [reply]
lgtm
